Assets/RedBlueGames/BulkRename/Editor/BulkRenamerWindow.cs

```diff
@@ -23,712 +23,755 @@
+
+		private BulkRenameGUIStyles guiStyles;
+		private BulkRenameGUIContents guiContents;
+		private Vector2 renameOperationsPanelScrollPosition;
+		private Vector2 previewPanelScrollPosition;
+
+		private BulkRenamer bulkRenamer;
+		private List<BaseRenameOperation> renameOperationsToClone;
+		private List<BaseRenameOperation> renameOperationsToApply;
+		private List<UnityEngine.Object> objectsToRename;
+		Dictionary<Object,string> allSpritesWithNewNames = new Dictionary<Object,string> ();
+
+
+		private List<UnityEngine.Object> ObjectsToRename {
+			get {
+				if (this.objectsToRename == null) {
+					this.objectsToRename = new List<UnityEngine.Object> ();
+				}
+
+				return this.objectsToRename;
+			}
+		}
+
+		private bool ShowDiff {
+			get {
+				return EditorPrefs.GetBool (ShowDiffPrefkey, true);
+			}
+
+			set {
+				EditorPrefs.SetBool (ShowDiffPrefkey, value);
+			}
+		}
+
+		[MenuItem (AssetsMenuPath, false, 1011)]
+		[MenuItem (GameObjectMenuPath, false, 49)]
+		private static void ShowRenameSpritesheetWindow ()
+		{
+			var bulkRenamerWindow = EditorWindow.GetWindow<BulkRenamerWindow> (true, "Bulk Rename", true);
+
+			// When they launch via right click, we immediately load the objects in.
+			bulkRenamerWindow.LoadSelectedObjects ();
+		}
+
+		private static bool ObjectIsValidForRename (UnityEngine.Object obj)
+		{
+			if (AssetDatabase.Contains (obj)) {
+				// Create -> Prefab results in assets that have no name. Typically you can't have Assets that have no name,
+				// so we will just ignore them for the utility.
+				return !string.IsNullOrEmpty (obj.name);
+			}
+
+			if (obj.GetType () == typeof(GameObject)) {
+				return true;
+			}
+
+			return false;
+		}
+
+		private static Texture GetIconForObject (UnityEngine.Object unityObject)
+		{
+			var pathToObject = AssetDatabase.GetAssetPath (unityObject);
+			Texture icon = null;
+			if (string.IsNullOrEmpty (pathToObject)) {
+				if (unityObject.GetType () == typeof(GameObject)) {
+					icon = EditorGUIUtility.FindTexture ("GameObject Icon");
+				} else {
+					icon = EditorGUIUtility.FindTexture ("DefaultAsset Icon");
+				}
+			} else {
+				if (unityObject is Sprite) {
+					icon = GetIconOfSprite (unityObject);
+				} else {
+					icon = AssetDatabase.GetCachedIcon (pathToObject);
+				}
+			}
+			return icon;
+		}
+
+		private	static Texture GetIconOfSprite (Object unityObject)
+		{
+			Sprite sprite = (Sprite)unityObject;
+			string path = AssetDatabase.GetAssetPath (sprite.texture);
+			TextureImporter ti = AssetImporter.GetAtPath (path) as TextureImporter;
+			bool isReadable = ti.isReadable;
+			ti.isReadable = true;
+			Texture texture = TextureFromSprite (sprite);
+			ti.isReadable = isReadable;
+			return texture;
+		}
+
+		/// <summary> ///trothmaster:  http://answers.unity3d.com/questions/651984/convert-sprite-image-to-texture.html /// </summary>
+		private static Texture2D TextureFromSprite (Sprite sprite)
```

Comment:
I really appreciate the link to the answers entry. Function name should be made to a verb. I recommend `CreateTextureFromSprite`.